src/System.Net.Http/tests/FunctionalTests/PostScenarioTest.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Net.Http.Headers;
+using System.Net.Tests;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Http.Functional.Tests
+{
+    // Note:  Disposing the HttpClient object automatically disposes the handler within. So, it is not necessary
+    // to separately Dispose (or have a 'using' statement) for the handler.
+    public class PostScenarioTest
+    {
+        private const string ExpectedContent = "Test contest";
+        private const string UserName = "user1";
+        private const string Password = "password1";
+        private readonly static Uri BasicAuthServerUri =
+            HttpTestServers2.BasicAuthUriForCreds(false, UserName, Password);
+        private readonly static Uri SecureBasicAuthServerUri =
+            HttpTestServers2.BasicAuthUriForCreds(true, UserName, Password);
+
+        readonly ITestOutputHelper _output;
+
+        public static object[][] EchoServers
+        {
+            get
+            {
+                return HttpTestServers2.EchoServers;
+            }
+        }
+
+        public static object[][] BasicAuthEchoServers
```

Comment:
This (and `EchoServers` above) could be a readonly static field, which would eliminate the penalty of repeated allocations on repeated calls to the get accessor.